shared/elements/gaia-header/dist/gaia-header.js

```diff
@@ -1127,8 +1127,9 @@ module.exports = component.register('gaia-header', {
   sumButtonWidths: function(buttons) {
     var defaultWidth = 50;
     var sum = buttons.reduce((prev, button) => {
+      var isEmpty = !button.children.length && !button.textContent;
```

Comment:
instead of children.length, it's likely faster to use "button.firstElementChild" or event "button.firstChild" (if it's null, it means there is no child). I think we want to call trim() on button.textContent too.